microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -177,20 +226,74 @@ def _install(self) -> bool:
             node.get_pure_path("~/.bashrc"),
             append=True,
         )
+
         return True
 
-    def _install_dependencies(self) -> None:
-        node = self.node
-        cwd = node.working_path
-        lspci = node.tools[Lspci]
-        device_list = lspci.get_device_list()
-        self.is_connect_x3 = any(
-            ["ConnectX-3" in dev.device_info for dev in device_list]
-        )
+    def set_version_info_from_source_install(
+        self, branch_identifier: str, matcher: Pattern[str]
+    ) -> None:
+        match = matcher.search(branch_identifier)
+        if not match or not match.group("major") or not match.group("minor"):
+            fail(
+                f"Could not determine dpdk version info from '{self._dpdk_source}'"
+                f" with id: '{branch_identifier}' using regex: '{matcher.pattern}'"
+            )
+        else:
+            major, minor = map(int, [match.group("major"), match.group("minor")])
+            self._dpdk_version_info = VersionInfo(major, minor)
+
+    def _load_drivers_for_dpdk(self) -> None:
+        self.node.log.info("Loading drivers for infiniband, rdma, and mellanox hw...")
         if self.is_connect_x3:
             mellanox_drivers = "mlx4_core mlx4_ib"
         else:
             mellanox_drivers = "mlx5_core mlx5_ib"
+        if isinstance(self.node.os, Ubuntu):
+            self.node.execute(
+                "modprobe -a rdma_cm rdma_ucm ",
+                cwd=self.node.working_path,
+                expected_exit_code=0,
+                expected_exit_code_failure_message=(
+                    "Could not load rdma cm/ucm drivers"
+                ),
+                sudo=True,
+            )
+        elif isinstance(self.node.os, Redhat):
+            if not self.is_connect_x3:
+                self.node.execute(
+                    f"dracut --add-drivers '{mellanox_drivers} ib_uverbs' -f",
+                    cwd=self.node.working_path,
+                    expected_exit_code=0,
+                    expected_exit_code_failure_message=(
+                        "Issue loading mlx and ib_uverb drivers into ramdisk."
+                    ),
+                    sudo=True,
+                )
+            self.node.execute(
+                f"modprobe -a  mlx4_en {mellanox_drivers}",
+                cwd=self.node.working_path,
+                expected_exit_code=0,
+                expected_exit_code_failure_message=(
+                    "Encountered issue loading infiniband and "
+                    "mellanox drivers (redhat)."
+                ),
+                sudo=True,
+            )  # add mellanox drivers
```

Comment:
How about else? If other distros are not supported, raise an unsupported exception.

Reply:
Why does the UnsupportedDistroException make the test class parse the OS version info? That seems like the os class should contain that already and you should just pass in node.os or something